alexandria/numeric/src/interp.cairo

```diff
@@ -0,0 +1,89 @@
+// One-dimensional linear interpolation for monotonically increasing sample points.
+use array::ArrayTrait;
+
+#[derive(Serde, Copy, Drop, PartialEq)]
+enum Interpolation {
+    Linear: (),
+    Nearest: (),
+    ConstantLeft: (),
+    ConstantRight: (),
+}
+
+#[derive(Serde, Copy, Drop, PartialEq)]
+enum Extrapolation {
+    Null: (),
+    Constant: (),
+}
+
+// Interpolate y(x) at x.
+// # Arguments
+// * `x` - The position at which to interpolate.
+// * `xs` - The sorted abscissa sequence of len L.
+// * `ys` - The ordinate sequence of len L.
+// * `interpolation` - The interpolation method to use.
+// * `extrapolation` - The extrapolation method to use.
+// # Returns
+// * `usize` - The interpolated y at x.
+fn interp(
+    x: usize,
+    xs: @Array<usize>,
+    ys: @Array<usize>,
+    interpolation: Interpolation,
+    extrapolation: Extrapolation
+) -> usize {
+    // [Check] Inputs
+    assert(xs.len() == ys.len(), 'Arrays must have the same len');
+    assert(xs.len() >= 2_usize, 'Array must have at least 2 elts');
+
+    // [Check] Extrapolation
+    if x < *xs.at(0) {
+        let y = match extrapolation {
+            Extrapolation::Null(()) => 0_usize,
+            Extrapolation::Constant(()) => *ys.at(0_usize),
+        };
+        return y;
+    }
+    if x > *xs.at(xs.len() - 1_usize) {
+        let y = match extrapolation {
+            Extrapolation::Null(()) => 0_usize,
+            Extrapolation::Constant(()) => *ys.at(xs.len() - 1_usize),
+        };
+        return y;
+    }
+
+    // [Compute] Interpolation
+    let mut index = 0_usize;
+    loop {
+        if index + 1_usize == xs.len() {
+            assert(0 != 1, 'Unexpected error');
```

Comment:
```suggestion
            assert(0 == 1, 'Unexpected error');
```
You mean that right ?

Reply:
I replaced it by `false`  the I just removed it since this case cannot be reached logically.